### ISSUE TITLE
Valo error type notifications missing close box and docs are wrong

### ISSUE THREAD

**Issue by cbmeeks** [vaadin/framework#8296]
First, the default behavior of the error notifications don't include a close box.  This is confusing our users because they think the screen is "frozen".  As a work-around, I've started including a small delay (setDelayMsec) to 10 seconds so that the error box no longer prevents them from at least using other components.

There needs to be a close box added back.  Or, at least the option to include one.

Second, the documentation is conflicting on the subject.

The docs I am reading are:

https://vaadin.com/docs8/-/part8/framework/application/application-notifications.html

![image](https://cloud.githubusercontent.com/assets/49341/22152588/9a95d26c-def1-11e6-810c-3af6e0b4efaf.png)

The docs still indicate a close box is there but the image (and the actual error notification) are missing the close box.

Finally, further down the page there is more conflicting information:

![image](https://cloud.githubusercontent.com/assets/49341/22152643/e4e2eb2a-def1-11e6-8c32-d514cf8dbd4d.png)

This also indicates there is a close box on the error notification.

**Comment by jouni** [comment]
You can add the close button yourself by adding an additional style name for the notification:

```java
notification.setStyleName("error closable");
```

For some reason, the `Notification` class only has `setStyleName` and no `addStyleName`, so you have to concatenate styles together in one string.

**Comment by jouni** [comment]
You are probably right, that the error type notification should have the close button visible by default. That changed in Valo, and I take the blame for that.

In hindsight, we should have introduced another style name for the visual looks of the error type, and the error type should have had the close button visible by default (like it is in all the other built-in themes).

I’m not against fixing that in Vaadin 8, but it is a breaking change in a way. If users are relying on the existing behavior, they would need to update their code to not use the error type, but instead modify the notification style name and duration manually.

**Comment by jouni** [comment]
So I’m a bit unsure whether it’s worth fixing. @Legioth, @Artur-, any opions?

**Comment by Legioth** [comment]
I'm in favor of changing the default in 8, as long as it doesn't require too much effort from the implementation team.

**Comment by jouni** [comment]
The effort would be to include one Sass mixin for the `.error` notification style…

_notification.scss
```scss
  .#{$primary-stylename}.error {
    @include valo-notification-error-style($primary-stylename);
    // Added
    @include valo-notification-closable-style;
  }
```

…and to decide on a name for the new style name that only applies the error notification type styles (proposing `.danger`), but not the closable style:

_notification.scss
```scss
  // Added
  .#{$primary-stylename}.danger {
    @include valo-notification-error-style($primary-stylename);
  }
```

**Comment by jouni** [comment]
Probably some screenshot tests will also break after this.

**Comment by Legioth** [comment]
@jouni Would you have time to make a pull request?

**Comment by phansson** [comment]
This problem was logged by me in April 2015 :   "**Valo theme: Error Notifications have no visual close indicator",** 
On old Issue Tracker:  https://dev.vaadin.com/ issue #17373
On GitHub Issue Tracker: ticket no #6587.
(note that screenshots seems to have been lost when going from old issue tracker to new)


I'm glad is is now fixed. Just wondering why it wasn't fixed back when it was reported against Vaadin 7 way back then?

**Comment by jouni** [comment]
> Just wondering why it wasn't fixed back when it was reported against Vaadin 7 way back then?

I personally admit not being in the loop about Framework issues after 2014 or something like that (as I was not part of the team after implementing Valo around 2013–2014), and pretty much dropping the ball on improving Valo after that.

But @Artur- luckily assigned me to this issue, and the team requested me to fix this before 8.0 final, and I agreed it being important enough and easy to fix so I dusted off my Eclipse and Framework skills to get this one fixed.

**Comment by phansson** [comment]
Jouni: I very much appreciate the honest reply.  🥇 
You guys in Vaadin Inc play with open cards and generally little filter and this resonates very well with me.
